events/code_review_events/workflow.py

```diff
@@ -397,6 +412,18 @@ def __init__(self, cache_root):
             self.bugbug_utils = None
             logger.info("Skipping workers consumers")
 
+    def handle_signal(self, *args):
+        if isinstance(self.workflow, CodeReview) is False:
+            return
+
+        logger.info("Events: received SIGTERM")
+        if self.workflow.cached_build() is not None:
+            logger.info(
+                "Events: having unprocessed build",
+                build=str(self.workflow.cached_build()),
+            )
+            self.workflow.process_build(self.workflow.cached_build())
```

Comment:
In SIGTERM handler, we have limited time to cleanup. Instead of trying to process the build, we should just put it back in the queue, so the next worker that boots up will process it.

Reply:
What does limited time mean? in POSIX there is no standard telling us that there is such a time between SIGTERM and SIGKILL or SIGSTOP. 
SIGTERM only signals our app to stop but since we don't intercept that signal we don't stop and that maybe the send message is sent.

Reply:
I think we should intercept SIGTERM, do the cleaning and terminate the ap ourselfs.

Reply:
Heroku sends a SIGTERM and gives you 30 seconds to cleanup.

Reply:
This is great then! But if we terminate the app in the callback do you think that there is the possibility of not consuming another message from the bus?

Reply:
It's OK if we don't consume the messages, as they will stay in the queue and the next worker that is going to be started is goint to consume them properly.